packages/babel-core/src/transformation/file/index.js

```diff
@@ -410,8 +412,32 @@ export default class File extends Store {
   }
 
   parse(code: string) {
+    let opts = this.opts;
+    let parseCode = parse;
+    if (opts.parserOpts.parser) {
+      parseCode = opts.parserOpts.parser;
+
+      if (typeof parseCode === "string") {
+        let dirname = opts.parserOpts.dirname || process.cwd();
+        let parser = resolve(parseCode, dirname);
+        if (parser) {
+          parseCode = require(parser).parse;
+        } else {
+          throw new Error(`Couldn't find parser ${parseCode} with "parse" method relative to directory ${dirname}`);
+        }
+      }
+
+      if (!this.parserOpts.parser) {
+        this.parserOpts.parser = {
```

Comment:
Good question actually - this.opts.parserOpts is the new option I added for `babelrc` config as in packages/babel-core/src/transformation/file/options/config.js

this.parserOpts are some defaults or something? - https://github.com/babel/babel/blob/7c23d97ae9ff8b360bc1c3b2f568a05633c704ce/packages/babel-core/src/transformation/file/index.js#L53-L59